sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py

```diff
@@ -155,6 +156,12 @@ def __init__(self, hostname, auth=None, **kwargs):
         self._receive_settle_mode = kwargs.pop('receive_settle_mode', ReceiverSettleMode.Second)
         self._desired_capabilities = kwargs.pop('desired_capabilities', None)
 
+        # transport
+        if kwargs.get('transport_type') is TransportType.Amqp and kwargs.get('http_proxy') is not None:
+            raise ValueError("Http proxy settings can't be passed if transport_type is explicitly set to Amqp")
+        self._transport_type = kwargs.pop('transport_type', TransportType.Amqp)
+        self._http_proxy = kwargs.pop('http_proxy', None)
```

Comment:
should the default transport type change if http_proxy is passed in? (same comment as above).

Reply:
same answer as above :P
the transport type will eventually be passed into transport where the this condition is checked. So, this won't be necessary